Add callout macro

domoscargin opened the pull request:
## What

Simplifies callouts and notes into a single callout macro. Used for branding stuff.

## Notes
- Previously, I special cased the WCAG callout, but I've reverted that as not worth the hassle - keeping the old component till we're ready to get rid of it just makes it easier to find and extract and offers less complication to this work.

owenatgov reviewed (dismissed):
💪🏻

The amount of stuff we're doing for the wcag callouts feels a bit weird but that's not a blocker for this. It makes me wonder about if we use this callout pattern more, how we go about managing repeated content and how generic/specific we want to make the macro. From this, I feel like more generic is better, probably.

domoscargin commented:
@owenatgov Speaking to @romaricpascal, I think we came down on the side of revisiting a "macro'd" way of displaying WCAG and brand content. I'll probably merge this for now, but the next iteration will generify everything a bit more.

romaricpascal reviewed (commented):
Thanks for reworking it to not worry about the WCAG stuff too much. Note a couple of little things (nothing major). 

I'm also wondering if the PR points to the right branch. Looks like we'll merge it on a testing branch, just want to check if that branch will be merged on `main` later 😊

domoscargin commented on src/components/back-link/index.md:
```diff
@@ -47,4 +47,4 @@ Use the `govuk-back-link--inverse` modifier class to show a white link on a dark
 
-{{ example({ group: "components", item: "back-link", example: "inverse", html: true, nunjucks: true, open: false }) }}
\ No newline at end of file
+{{ example({ group: "components", item: "back-link", example: "inverse", html: true, nunjucks: true, open: false }) }}
```
I had a look locally, and it's adding a newline to the end of the file. I'm not sure why that's not reflected here. I can save these files without formatting to avoid ambiguity if preferred?

domoscargin commented on src/stylesheets/components/_callout.scss:
```diff
@@ -0,0 +1,37 @@
+.app-callout .app-callout__heading {
+  @extend %govuk-heading-m;
+  max-width: 30em;
+}
```
Not intentional!

We do need the combined classes to "out-specify" `.app-content h2` which gets set later on.

I've reduced down to one selector now.

domoscargin commented:
@romaricpascal Should've addressed all your comments now.

It's the correct base branch - I guess at some point it evolved from a testing branch to the actual working branch!

domoscargin commented:
I'm going to merge this now, we can tweak on the doc PR.
